src/api/song.js

```diff
@@ -8,18 +8,18 @@ router.use(hostHandler)
 router.post('/', async (req, res) => {
     const host = req.sessionHost
     const { songId } = req.body
-    if (!songId || songId.slice(0, 14) !== "spotify:track:") {
-        const { statusCode, body: song } = await host.spotifyApi.getSongById(songId.slice(14))
-        if (statusCode !== 200) {
-            return res.status(400).send('Song id not found')
-        }
-        if (host.songQueue.includes(song)) {
-            return res.status(400).send('Song already in the queue')
-        }
-        return res.json(host.addSong(song))
-    } else {
+    if (!songId) {
         return res.status(400).send('Invalid input')
     }
+    
+    const { statusCode, body: song } = await host.spotifyApi.getSongById(songId)
+    if (statusCode !== 200) {
```

Comment:
Now that we're refactoring this I think that we should check the actual "not found" code here (prob. 404) and otherwise catch the error and forward it as 500 "Something went wrong" to the user

Reply:
Seems to be 400 actually. From https://developer.spotify.com/console/get-track/?id=3n3Ppam7vgaVa1iaRUc9&market= (with a messed up song id):

![image](https://user-images.githubusercontent.com/35035388/61992577-fed57480-b068-11e9-8d13-0cea2720379d.png)